### DISCUSSION TITLE
F´ Release v3.1.0

### DISCUSSION THREAD

**Discussion by LeStarch** [nasa/fprime#1549] [Announcements]
# F´ Release v3.1.0: Release Notes

This release of F´ publishes and baselines native versions of the FPP tools as well as a streamlined CMake system that should enable F´ builds based on FPP to run faster.  It includes a number of other improvements including a refined dependency versioning setup, `gcovr` for coverage analysis, and more!

## Major Changes
- Improved FPP and CMake efficiency. 
- Switched to `requirements.txt` for dependency version tracking.  Users should use this to install.
- ActiveRateGroup, RateGroupDriver now have zero-argument standardized constructors. Users should update topologies. See Ref App.
- CMake autocoders and targets have been refactored for efficiency. Implements should follow: [Autocoder Integration](https://github.com/nasa/fprime/blob/devel/docs/UsersGuide/dev/autocoder-integration.md)

## New Contributors
* @capsulecorplab made their first contribution in https://github.com/nasa/fprime/pull/1198
* @myint made their first contribution in https://github.com/nasa/fprime/pull/1272
* @iw4p made their first contribution in https://github.com/nasa/fprime/pull/1240
* @danjwait made their first contribution in https://github.com/nasa/fprime/pull/1346
* @Drosaca made their first contribution in https://github.com/nasa/fprime/pull/1352
* @inPhraZ made their first contribution in https://github.com/nasa/fprime/pull/1322
* @Kronos3 made their first contribution in https://github.com/nasa/fprime/pull/1383
* @lnxpy made their first contribution in https://github.com/nasa/fprime/pull/1353
* @cadomani made their first contribution in https://github.com/nasa/fprime/pull/1423
* @connorfuhrman made their first contribution in https://github.com/nasa/fprime/pull/1441
* @aditsachde made their first contribution in https://github.com/nasa/fprime/pull/1508

**Full Changelog**: https://github.com/nasa/fprime/compare/v3.0.0...v3.1.0

<hr /><em>This discussion was created from the release <a href='https://github.com/nasa/fprime/releases/tag/v3.1.0'>F´ Release v3.1.0</a>.</em>

**Comment by LeStarch** [comment]
Note: we changed the default branch to `master` as most users of F´ should be off this more stable branch.  Developers should still use `devel` and make PRs to `devel`.

This seeks to resolve the issue where users cloning F´  where seeing some of the churn in `devel` where the published documentation wasn't always inline with the development root.

**Comment by ThibFrgsGmz** [comment]
I just saw that the `CHANGELOG.md` has not been updated. Should it be considered deprecated and left there for legacy?

I personally don't mind, it seems better to use the GitHub functionality and it would be easier to maintain.
